### DISCUSSION TITLE
Little navmap compilation error related to atools

### DISCUSSION THREAD

**Discussion by Mohammad-Imran01** [albar965/littlenavmap#1064] [General]
![image](https://github.com/albar965/littlenavmap/assets/117526105/5b6291fd-aea3-4ce5-be3d-75ece87259d7)

**Comment by albar965** [comment]
What Qt version do you use to compile?
Can you provided the whole compilation log in text form?

**Reply by Mohammad-Imran01** [reply]
I was using  5.12.9 but then i changed it to 5.15.2 the issues are gone but now i face marble issue which i am unable to install for 10 days

**Reply by Mohammad-Imran01** [reply]
C:\QtProjects\littlenavmap\src\common\coordinateconverter.h:21: error: marble/GeoDataCoordinates.h: No such file or directory
In file included from ..\littlenavmap\src\common\coordinateconverter.cpp:18:
..\littlenavmap\src/common/coordinateconverter.h:21:10: fatal error: marble/GeoDataCoordinates.h: No such file or directory
 #include <marble/GeoDataCoordinates.h>
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
:-1: error: [Makefile:12285: coordinateconverter.o] Error 1
C:\QtProjects\littlenavmap\src\gui\mainwindow.h:28: error: marble/MarbleGlobal.h: No such file or directory
In file included from ..\littlenavmap\src\airspace\airspacecontroller.cpp:31:
..\littlenavmap\src/gui/mainwindow.h:28:10: fatal error: marble/MarbleGlobal.h: No such file or directory
 #include <marble/MarbleGlobal.h>
          ^~~~~~~~~~~~~~~~~~~~~~~
C:\QtProjects\littlenavmap\src\common\aircrafttrail.cpp:34: error: marble/GeoDataLatLonAltBox.h: No such file or directory
..\littlenavmap\src\common\aircrafttrail.cpp:34:10: fatal error: marble/GeoDataLatLonAltBox.h: No such file or directory
 #include <marble/GeoDataLatLonAltBox.h>
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
C:\QtProjects\littlenavmap\src\gui\mainwindow.h:28: error: marble/MarbleGlobal.h: No such file or directory
In file included from ..\littlenavmap\src\app\navapp.cpp:35:
..\littlenavmap\src/gui/mainwindow.h:28:10: fatal error: marble/MarbleGlobal.h: No such file or directory
 #include <marble/MarbleGlobal.h>
          ^~~~~~~~~~~~~~~~~~~~~~~
![Uploading Screenshot (27).png…]()

**Reply by albar965** [reply]
You need to build Marble from the branch `lnm/1.1`. Scripts for Windows are here: https://github.com/albar965/marble/tree/lnm/1.1/build/win
Start with this https://github.com/albar965/marble/blob/lnm/1.1/build/win/build_release_64.cmd or this https://github.com/albar965/marble/blob/lnm/1.1/build/win/build_debug_64.cmd .

I use these cmd scripts to build the official releases. They can give you a hint about how to build it.

Alex

**Reply by Mohammad-Imran01** [reply]
Thanks alot for your responses. Okay let me try.

**Reply by Mohammad-Imran01** [reply]
Hi i dont understand what to do with these cmd file command before or after it can u elaborate a little bit. I have always tried this (https://github.com/albar965/littlenavmap/wiki/Compiling) but can't compile with any qt version

**Reply by albar965** [reply]
LNM Marble should compile with Qt 5.15.2

**Reply by Mohammad-Imran01** [reply]
See it gives these errors or other error with other version but i cannot install it
[marbleInstallErrors.txt](https://github.com/albar965/littlenavmap/files/13502219/marbleInstallErrors.txt)
![Screenshot (28)](https://github.com/albar965/littlenavmap/assets/117526105/1c551b9f-36a5-4720-87a7-9dd0b24751f9)
![Screenshot (29)](https://github.com/albar965/littlenavmap/assets/117526105/22b05d5a-24b1-4e66-8946-ccb4f80114fb)

**Reply by Mohammad-Imran01** [reply]
Check line 453 in txt file

**Reply by albar965** [reply]
No idea why it cannot find `QApplication`. Do you really want to build the 32-bit version?
Is the environment variable `%APROJECTS%` set to the path containing all projects like `Marble`, `atools` and `littlenavmap`?

**Reply by Mohammad-Imran01** [reply]
No any 64 or 32 both will work, I dont have a folder APROJECT

**Reply by albar965** [reply]
Not a folder `APROJECT` but an **environment variable** `APROJECTS`

**Reply by Mohammad-Imran01** [reply]
![Screenshot (30)](https://github.com/albar965/littlenavmap/assets/117526105/a891943b-4a53-4c71-83e2-a364dc79350b)
![Screenshot (31)](https://github.com/albar965/littlenavmap/assets/117526105/4c670b0a-0901-4ad1-bdad-d3f90d9dfe93)
![Screenshot (33)](https://github.com/albar965/littlenavmap/assets/117526105/e1c57f5b-b99e-4484-87da-b643aec0e2ce)

**Comment by albar965** [comment]
APROJECT **_S_**

**Reply by albar965** [reply]
You're missing an `S`.

**Reply by Mohammad-Imran01** [reply]
![Screenshot (34)](https://github.com/albar965/littlenavmap/assets/117526105/b9aca300-ba7d-4c30-b34a-e3738fdc16c4)
![Screenshot (35)](https://github.com/albar965/littlenavmap/assets/117526105/7361b9df-5514-4980-9b14-3148a1789636)

**Reply by albar965** [reply]
APROJECTS is set wrongly.
Set `APROJECTS` to `C:\Projekte` (or whatever base folder your use) in the environment variables.
Then put `atools`, `Marble` and `littlenavmap` into `C:\Projekte`.

**Reply by Mohammad-Imran01** [reply]
![Screenshot (37)](https://github.com/albar965/littlenavmap/assets/117526105/b0477bec-9f1c-4705-b6fe-7df42672c9fe)

**Reply by Mohammad-Imran01** [reply]
Please have a look.
Should i run these commands now

cd C:\Projekte\build-marble-release
set PATH=%PATH%;C:\Qt\5.15.2\mingw81_32\bin;C:\Qt\Tools\mingw810_32\bin
cmake -Wno-dev -G "MinGW Makefiles" -DQTONLY=TRUE -DQT5BUILD=TRUE -DWITH_quazip=FALSE -DWITH_libshp=FALSE -DWITH_libgps=FALSE -DWITH_libwlocate=FALSE -DWITH_QtLocation=FALSE -DWITH_liblocation=FALSE -DCMAKE_INSTALL_PREFIX="C:\Projekte\Marble-release" -DCMAKE_BUILD_TYPE=Release ..\marble
mingw32-make -j2
mingw32-make install

**Reply by albar965** [reply]
Looking good so far. :+1: 
Remove ATOOLS_NO_FS,  ATOOLS_NO_GRIB. Otherwise it won't compile.

**Reply by Mohammad-Imran01** [reply]
![Screenshot (38)](https://github.com/albar965/littlenavmap/assets/117526105/1b3138ca-5230-42ac-804b-532fdd1fc836)
Please have a look.....

**Reply by albar965** [reply]
Look ok but this the path to the 32-bit MinGW.

**Reply by Mohammad-Imran01** [reply]
Removed 32bit, leaving 64bit mingw path only

**Reply by Mohammad-Imran01** [reply]
Now

**Reply by Mohammad-Imran01** [reply]
See same issue:
[marbleInstallingIssue.txt](https://github.com/albar965/littlenavmap/files/13504064/marbleInstallingIssue.txt)

**Reply by albar965** [reply]
You're using the wrong cmake configuration which tries to build apps, test files and whatnot.

Again: Look at the .cmd files to see how to build it: 
https://github.com/albar965/marble/blob/564a141497046946b5b9903336e1ab12cdf9049f/build/win/build_base.cmd#L25

For example:
```
cmake -Wno-dev -G "MinGW Makefiles" -DCMAKE_BUILD_TYPE=%BUILDTYPE% -DSTATIC_BUILD=TRUE -DQTONLY=TRUE -DBUILD_MARBLE_EXAMPLES=NO -DBUILD_INHIBIT_SCREENSAVER_PLUGIN=NO -DBUILD_MARBLE_APPS=NO -DBUILD_MARBLE_EXAMPLES=NO -DBUILD_MARBLE_TESTS=NO -DBUILD_MARBLE_TOOLS=NO -DBUILD_TESTING=NO -DBUILD_WITH_DBUS=NO -DMARBLE_EMPTY_MAPTHEME=YES -DMOBILE=NO -DWITH_DESIGNER_PLUGIN=NO -DWITH_Phonon=NO -DWITH_Qt5Location=NO -DWITH_Qt5Positioning=NO -DWITH_Qt5SerialPort=NO -DWITH_ZLIB=NO -DWITH_libgps=NO -DWITH_libshp=NO -DWITH_libwlocate=NO -DCMAKE_INSTALL_PREFIX=%DEPLOYDIR% ..\marble\
```

**Reply by Mohammad-Imran01** [reply]
1. Create three more variables?
if defined APROJECTS ( echo %APROJECTS% ) else ( echo APROJECTS not set && exit /b 1 )
if defined DEPLOYDIR ( echo %DEPLOYDIR% ) else ( echo DEPLOYDIR not set && exit /b 1 )
if defined BUILDDIR ( echo %BUILDDIR% ) else ( echo BUILDDIR not set && exit /b 1 )
if defined BUILDTYPE ( echo %BUILDTYPE% ) else ( echo BUILDTYPE not set && exit /b 1 )


2. Run the cmd file... right?

**Reply by albar965** [reply]
> Create three more variables?

No this is done by the other scripts which call this one.

> Run the cmd file... right?

Run one of the [build_debug_32.cmd](https://github.com/albar965/marble/blob/564a141497046946b5b9903336e1ab12cdf9049f/build/win/build_debug_32.cmd), [build_debug_64.cmd](https://github.com/albar965/marble/blob/564a141497046946b5b9903336e1ab12cdf9049f/build/win/build_debug_64.cmd), [build_release_32.cmd](https://github.com/albar965/marble/blob/564a141497046946b5b9903336e1ab12cdf9049f/build/win/build_release_32.cmd), [build_release_64.cmd](https://github.com/albar965/marble/blob/564a141497046946b5b9903336e1ab12cdf9049f/build/win/build_release_64.cmd)
scripts which call the one I've linked to show that you use the wrong build options.

**Reply by Mohammad-Imran01** [reply]
Thanks alooooooooooooooooooooooooot...................................... I think marbel is done do i need to move files ????????
![Screenshot (40)](https://github.com/albar965/littlenavmap/assets/117526105/0846aa16-bfa4-4484-98e4-5bc1f8d9d41c)

**Reply by albar965** [reply]
Yay. :fireworks:  This should be good to compile LNM 64-bit.

**Reply by Mohammad-Imran01** [reply]
So i just need to open LNM and atools in QT, Set it to mingw5.15.2 release mingw32 for both of them, Set atools as a dependency for LNM then compile LNM and its done? Is mingw64 necessary becauase then i have to install it

**Reply by albar965** [reply]
Is there a folder `Marble-release` in the folder `Projekte` after compiling Marble? The Marble cmd scripts should create it.

You have to use the same mingw path as the Marble build to build atools and LNM. This is `C:\Qt\5.15.2\mingw81_64\bin\` and `C:\Qt\Tools\mingw810_64\bin` (might differ from your installation)

**Reply by Mohammad-Imran01** [reply]
not exactly.....
![Screenshot (41)](https://github.com/albar965/littlenavmap/assets/117526105/088f3348-8c8b-47a9-9dd6-6162a0aa2522)

**Reply by albar965** [reply]
No, it's ok, sry. I was looking at my Linux build folder here. You screenshot shows how it has to look on Windows.

**Reply by Mohammad-Imran01** [reply]
Sorry but i didnt had 64bit in QT i just downloaded it now, for installing it used 32bit only i guess.
![Screenshot (42)](https://github.com/albar965/littlenavmap/assets/117526105/66b97bc2-5ce5-4106-b595-c9263aa7f402)

**Reply by albar965** [reply]
Whatever version you'd like to build. 

But keep the 64/32-bit and debug/release builds in sync. Otherwise it will not compile.

**Reply by Mohammad-Imran01** [reply]
32bit will be better because my project works with 32bit-mingw only

**Reply by albar965** [reply]
Then you have to rebuild Marble with `build_debug_32.cmd` or `build_release_32.cmd`.

**Reply by Mohammad-Imran01** [reply]
Okay.... I am doing it.

**Reply by Mohammad-Imran01** [reply]
there is a problem: 
![Screenshot (44)](https://github.com/albar965/littlenavmap/assets/117526105/84ad7c2b-240c-460d-834c-033737b30664)
![Screenshot (45)](https://github.com/albar965/littlenavmap/assets/117526105/609ccddc-d88e-4afc-b516-fd4e8faa736d)

**Reply by albar965** [reply]
Call make deploy before running it in Qt Creator.
This is my run config:
![grafik](https://github.com/albar965/littlenavmap/assets/15254956/6e452b88-6d65-4628-8807-9694ddfd126b)

Also make sure that Marble and everything else is 32-bit debug.

**Reply by Mohammad-Imran01** [reply]
I am unable to follow this..
Can you please elaborate


 
Let me show you my current directories and variables

![Screenshot (53)](https://github.com/albar965/littlenavmap/assets/117526105/4b9a80e4-baeb-4b3f-b242-3dfa98dcbca6)
![Screenshot (52)](https://github.com/albar965/littlenavmap/assets/117526105/30406f6e-88b1-4121-a115-c6ad05b04a17)
![Screenshot (51)](https://github.com/albar965/littlenavmap/assets/117526105/7ed026f2-7db9-4fc6-bb1f-75513e24758f)
![Screenshot (49)](https://github.com/albar965/littlenavmap/assets/117526105/036dc61b-6d75-4a06-80fc-6325c8c2ca5b)
![Screenshot (50)](https://github.com/albar965/littlenavmap/assets/117526105/58550bb5-88ce-4246-b253-19ab9cd9a435)

**Comment by Mohammad-Imran01** [comment]
The Error is:
![Screenshot (54)](https://github.com/albar965/littlenavmap/assets/117526105/34bac84d-6107-4a7e-8822-6a1572a27f5a)

**Reply by Mohammad-Imran01** [reply]
![Screenshot (55)](https://github.com/albar965/littlenavmap/assets/117526105/24da02ac-c26b-4405-bbbf-d2996467bc64)

**Reply by Mohammad-Imran01** [reply]
![Screenshot (56)](https://github.com/albar965/littlenavmap/assets/117526105/cc25d9a4-468c-42d9-b5c7-d0dec40d9e17)

**Reply by Mohammad-Imran01** [reply]
Sorry for flooding here with so many screenshots........
![Screenshot (57)](https://github.com/albar965/littlenavmap/assets/117526105/69f39bb3-6b8d-43bb-8806-c2f469e3e52e)

**Reply by albar965** [reply]
Quickly glancing over this: Your deployment configuration is wrong. You need to call `mingw32-make.exe deploy` to build the running folder. This is huge project. Building is not expected to be easy.

And sorry, but my time is limited. I'm trying to get a release done and giving support in all kind of channels everywhere in parallel.

**Reply by Mohammad-Imran01** [reply]
How to fix it

**Reply by Mohammad-Imran01** [reply]
How did you changed the make command for deployment
